### PR TITLE
render most warnings fatal

### PR DESCRIPTION
**Pull request by jrochel**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,8 +12,8 @@ include Makefile.options
 ##			      Internals
 
 ## Required binaries
-ELIOMC            := eliomc
-ELIOMOPT          := eliomopt
+ELIOMC            := eliomc ${WARNING_FLAGS}
+ELIOMOPT          := eliomopt ${WARNING_FLAGS}
```
Comment (ilankri) [inline Makefile:16]: Shouldn't we do the same change for `JS_OF_ELIOM`?

File: src/os_user.eliomi [hunk 2]
```diff
@@ -119,9 +119,6 @@ val is_complete : Os_types.User.t -> bool
 
 [%%server.start]
 
-val emails_of_user : Os_types.User.t -> string Lwt.t
-(** [emails_of_user user] returns the emails of the user [user]. *)
-
```
Comment (ilankri) [inline src/os_user.eliomi:120]: Unwanted change?

File: src/os_user_view.eliomi [hunk 3]
```diff
@@ -286,12 +286,6 @@ val disconnect_all_link
     @param text_link link text.
  *)
 
-val disconnect_button
-  :  ?text_logout:string
-  -> unit
-  -> [> Html_types.li_content_fun] Eliom_content.Html.F.elt
-(** A disconnect button *)
-
```
Comment (ilankri) [inline src/os_user_view.eliomi:294]: Unwanted change?

File: src/os_user.eliomi (hunk 2 above)
Comment (jrochel) [reply src/os_user.eliomi:120]: No, the signature appeared twice in the same file.

File: src/os_user_view.eliomi (hunk 3 above)
Comment (jrochel) [reply src/os_user_view.eliomi:294]: No, the signature appeared twice in the same file.

File: src/os_session.eliomi [hunk 4]
```diff
@@ -22,57 +22,60 @@
...
     Use this scopes for example when you want to store
     server side data for one browser or tab, but not user dependent.
     (Remains when user logs out).
 *)
-val user_indep_state_hierarchy : Eliom_common.scope_hierarchy
-val user_indep_process_scope : [> Eliom_common.client_process_scope ]
-val user_indep_session_scope : [> Eliom_common.session_scope ]
+
+val user_indep_process_scope : [> Eliom_common.client_process_scope]
+val user_indep_session_scope : [> Eliom_common.session_scope]
```
[the remark this answers is not in the document]
Comment (ilankri) [reply src/os_session.eliomi:69]: But the comment is related to the three signatures (not only the first one).

File: Makefile (hunk 1 above)
Comment (jrochel) [reply Makefile:16]: Effectivement. C'est fait.

File: src/os_session.eliomi (hunk 4 above)
[the remark this answers is not in the document]
Comment (jrochel) [reply src/os_session.eliomi:69]: Ah oui. C'est réglé.

**Review by ilankri — APPROVED** [review]

File: src/os_session.eliomi (hunk 4 above)
[the remark this answers is not in the document]
Comment (ilankri) [reply src/os_session.eliomi:69]: Je ne vois pas de changement…

[the remark this answers is not in the document]
Comment (jrochel) [reply src/os_session.eliomi]: Pardon, j'ai oublié de pusher.

**Review by ilankri — APPROVED** [review]

**Pull request merged**